pkg/ovsdb/transact.go

```diff
@@ -411,6 +417,55 @@ func (mapUUID MapUUID) ResolvRow(row *map[string]interface{}) error {
 	return nil
 }
 
+type Etcd struct {
+	Cli  *clientv3.Client
+	Ctx  context.Context
+	If   []clientv3.Cmp
+	Then []clientv3.Op
+	Else []clientv3.Op
+	Res  *clientv3.TxnResponse
+}
+
+func NewEtcd(parent *Etcd) *Etcd {
+	return &Etcd{
+		Ctx: parent.Ctx,
+		Cli: parent.Cli,
+	}
+}
+func (etcd *Etcd) Clear() {
+	etcd.If = []clientv3.Cmp{}
+	etcd.Then = []clientv3.Op{}
+	etcd.Else = []clientv3.Op{}
+}
+
+func (etcd Etcd) String() string {
+	return fmt.Sprintf("if size(%d) then size(%d) else size(%d)", len(etcd.If), len(etcd.Then), len(etcd.Else))
+}
+
+func (etcd *Etcd) Commit() error {
+	res, err := etcd.Cli.Txn(etcd.Ctx).If(etcd.If...).Then(etcd.Then...).Else(etcd.Else...).Commit()
+	if err != nil {
+		return err
+	}
+	etcd.Res = res
+	return nil
+}
+
+func (etcd *Etcd) Split() []*Etcd {
+	ETCD_MAX_TXN_OPS := 128
```

Comment:
Pls, put it as a separate constant, if we work with splitting tnx, we probably will want to configure it.